azkaban-common/src/main/java/azkaban/imagemgmt/permission/PermissionManagerImpl.java

```diff
@@ -0,0 +1,148 @@
+ * permissions are defined on image type. The permissions are created using user role and access
+ * type. This class internally accesses image type cache to get image type metadata.
+ */
+@Singleton
+public class PermissionManagerImpl implements PermissionManager {
+
+  private static final Logger log = LoggerFactory.getLogger(PermissionManagerImpl.class);
+
+  private final ImageTypeDao imageTypeDao;
+
+  /**
+   * Permission for image type owner of role ADMIN. ADMIN owner has complete access to image
+   * management APIs including add/delete of new member as image type owner.
+   */
+  private static final Permission ADMIN_PERMISSION = new Permission(Type.ADMIN);
+  /**
+   * Permission for image type owner of role MEMBER. MEMBER owner has complete access to image
+   * management APIs except add/delete of new member as image type owner.
+   */
+  private static final Permission MEMBER_PERMISSION = new Permission(Type.CREATE, Type.GET,
+      Type.UPDATE, Type.DELETE);
+  /**
+   * The default permission is only GET access to the image management APIs.
+   */
+  private static final Permission READ_ONLY_PERMISSION = new Permission(Type.GET);
+
+  @Inject
+  public PermissionManagerImpl(final ImageTypeDao imageTypeDao) {
+    this.imageTypeDao = imageTypeDao;
+  }
+
+  /**
+   * Checks the permission based on image type name, user id and Permission type.
+   *
+   * @param imageTypeName
+   * @param userId
+   * @param type
+   * @return boolean
+   */
+  @Override
+  public boolean hasPermission(final String imageTypeName, final String userId, final Type type) {
+    // Gets the image type metadata including ownerships.
+    final Optional<ImageType> optionalImageType =
+        this.imageTypeDao.getImageTypeWithOwnershipsByName(imageTypeName);
+    boolean hasPermission = false;
+    if (optionalImageType.isPresent()) {
+      final ImageType imageType = optionalImageType.get();
+      if (!CollectionUtils.isEmpty(imageType.getOwnerships())) {
+        // Gets the permission of the user based on image type ownership metadata.
+        final Optional<Permission> optionalPermission = getPermission(userId,
+            imageType.getOwnerships());
+        // Check if the given Permission.Type contains in the permission of the user.
+        // Check if the the user permission contains Permission.Type ADMIN.
+        if (optionalPermission.isPresent() && (optionalPermission.get().isPermissionSet(type)
+            || optionalPermission.get().isPermissionSet(Permission.Type.ADMIN))) {
+          hasPermission = true;
+        }
+      } else {
+        log.error(String.format("API access permission check failed. There is no ownership record "
+            + "for image type: %s.", imageTypeName));
+        throw new ImageMgmtException(String.format("API access permission check failed. There is "
+            + "no ownership record for image type: "
+            + "%s.", imageTypeName));
+      }
+    } else {
+      log.error(
```

Comment:
Here also you should return false.

Reply:
Throwing the appropriate error message and that will eventually be API response.